internal/pkg/log/logformat.go

```diff
@@ -0,0 +1,25 @@
+package log
+
+import (
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+type LogFormat string
+
+const (
+	LogFormatConsole LogFormat = "console"
+	LogFormatJSON LogFormat = "json"
+)
+
+// Creates LogFormat from string.
+// On invalid value Console is used as default with an error.
+func NewLogFormat(format string) (LogFormat, error) {
+	switch format {
+	case "console":
+		return LogFormatConsole, nil
+	case "json":
+		return LogFormatJSON, nil
+	default:
+		return LogFormatConsole, errors.New(`log format must be "console" or "json"`)
+	}
```

Comment:
Is there some way to simplify this? I'm unsure if there is a way to use `LogFormat(format)` - that syntax doesn't seem to return an "ok" value to detect invalid input value.

Reply:
You can remove `NewLogFormat` method at all.

Right place for validation is here:
https://github.com/keboola/keboola-as-code/blob/ea0fa61c4e81094bf448240bffb50a8bb3116684/internal/pkg/log/encoder.go#L42-L44

There is a `panic` call.
- Panic is OK for programmer errors (if user cannot affect the value), as I wrote sooner, but here it is an user input.
- So you should return the error (and also at all higher levels).
- Is there any problem to do it this way? If so, I'll be happy to help.

Reply:
I think we have a misunderstanding. The string passed into NewLogFormat is a user input. User can use `--log-format=unknown`. That's what the validation in NewLogFormat is for.

Reply:
```
type Config struct {
  LogFormat LogFormat `validate:"one_of=console json"`
}
```

Reply:
`LogFormat.Validate() error`

```
func (v LogFormat) Validate() error {
  switch v {
    case Console, Json: return nil
    default return errors.New()
  }
}

func NewLogFormat(s string) error {
  switch LogFormat(s) {
    case Console, Json: return nil
    default return errors.New()
  }
}

LogFormat("foo")
```